crates/sargon/src/wrapped_radix_engine_toolkit/low_level/manifest_summary/account_deposit/simple_non_fungible_resource_bounds.rs

```diff
@@ -0,0 +1,195 @@
+use crate::prelude::*;
+
+/// Represents the bounds for a simple non-fungible resource, which can be either exact or not exact.
+#[derive(Clone, Debug, PartialEq, Eq, uniffi::Enum)]
+pub enum SimpleNonFungibleResourceBounds {
+    Exact {
+        amount: Decimal,
+        certain_ids: Vec<NonFungibleLocalId>,
+    },
+    NotExact {
+        certain_ids: Vec<NonFungibleLocalId>,
+        lower_bound: LowerBound,
+        upper_bound: UpperBound,
+        allowed_ids: AllowedIds,
+    },
+}
+
+impl SimpleNonFungibleResourceBounds {
+    pub fn exact(
+        amount: Decimal,
+        certain_ids: impl IntoIterator<Item = NonFungibleLocalId>,
+    ) -> Self {
+        Self::Exact {
+            amount,
+            certain_ids: certain_ids.into_iter().collect(),
+        }
+    }
+
+    pub fn not_exact(
+        certain_ids: impl IntoIterator<Item = NonFungibleLocalId>,
+        lower_bound: LowerBound,
+        upper_bound: UpperBound,
+        allowed_ids: AllowedIds,
+    ) -> Self {
+        Self::NotExact {
+            certain_ids: certain_ids.into_iter().collect(),
+            lower_bound,
+            upper_bound,
+            allowed_ids,
+        }
+    }
+
+    pub fn certain_ids(&self) -> Vec<NonFungibleLocalId> {
+        match self {
+            Self::Exact { certain_ids, .. } => certain_ids.clone(),
+            Self::NotExact { certain_ids, .. } => certain_ids.clone(),
+        }
+    }
+}
+
+impl From<ScryptoSimpleNonFungibleResourceBounds>
+    for SimpleNonFungibleResourceBounds
+{
+    fn from(value: ScryptoSimpleNonFungibleResourceBounds) -> Self {
+        match value {
+            ScryptoSimpleNonFungibleResourceBounds::Exact {
+                amount,
+                certain_ids,
+            } => Self::Exact {
+                amount: amount.into(),
+                certain_ids: certain_ids
+                    .into_iter()
+                    .map(NonFungibleLocalId::from)
+                    .collect(),
+            },
+            ScryptoSimpleNonFungibleResourceBounds::NotExact {
+                certain_ids,
+                lower_bound,
+                upper_bound,
+                allowed_ids,
+            } => Self::NotExact {
```

Comment:
change to using the constructor you have declared: `fn not_exact`, to which you add a validation of bounds.